Modules/Microsoft365DSC/DSCResources/MSFT_O365SearchAndIntelligenceConfigurations/MSFT_O365SearchAndIntelligenceConfigurations.psm1

```diff
@@ -0,0 +1,444 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.Boolean]
+        $ItemInsightsIsEnabledInOrganization,
+
+        [Parameter()]
+        [System.String]
+        $ItemInsightsDisabledForGroup,
+
+        [Parameter()]
+        [System.Boolean]
+        $PersonInsightsIsEnabledInOrganization,
+
+        [Parameter()]
+        [System.String]
+        $PersonInsightsDisabledForGroup,
+
+        [Parameter()]
```

Comment:
If IsSingleInstance is added, there is no need for the Ensure parameter anymore. It can be removed, including all code that came with it.